### PR TITLE
Override insecure when endpoint URL is set

### PR DESCRIPTION
**Pull request by sevaorlov**
When an endpoint is set in both Environment variable with "http" and passed in WithEndpointURL with "https", Insecure is set to true while the endpoint is used from WithEndpointURL.

Example
- OTEL_EXPORTER_OTLP_ENDPOINT is set to "http://env.endpoint/prefix" 
- WithEndpointURL is passed "https://someendpoint/somepath"

The real endpoint used is "http://someendpoint/somepath", which is actually neither of both.

### CODE REVIEW

File: exporters/otlp/otlpmetric/otlpmetrichttp/internal/oconf/options.go [hunk 1]
```diff
@@ -287,9 +287,7 @@ func WithEndpointURL(v string) GenericOption {
 
 		cfg.Metrics.Endpoint = u.Host
 		cfg.Metrics.URLPath = u.Path
-		if u.Scheme != "https" {
-			cfg.Metrics.Insecure = true
-		}
+		cfg.Metrics.Insecure = u.Scheme != "https"
```
Comment (pellared) [inline exporters/otlp/otlpmetric/otlpmetrichttp/internal/oconf/options.go:290]: See the header of the file

```
// Code created by gotmpl. DO NOT MODIFY.
// source: internal/shared/otlp/otlpmetric/oconf/options.go.tmpl
```

**Review by pellared — COMMENTED** [review]
Can you add a changelog entry?
Can you fix the same issue in `olptrace` and `otlplog` as well?

File: exporters/otlp/otlpmetric/otlpmetrichttp/internal/oconf/options.go (hunk 1 above)
Comment (sevaorlov) [reply exporters/otlp/otlpmetric/otlpmetrichttp/internal/oconf/options.go:290]: I also realised there is the same issue for grpc exporter. Yes, I can do all fixes. Sorry, haven't checked the header :)

Comment (pellared) [reply exporters/otlp/otlpmetric/otlpmetrichttp/internal/oconf/options.go:290]: No problem. Thanks for the PR 👍

Comment (sevaorlov) [reply exporters/otlp/otlpmetric/otlpmetrichttp/internal/oconf/options.go:290]: looks like otlplog doesn't have same issue https://github.com/open-telemetry/opentelemetry-go/blob/506a9baf5e00147118caea839b529e0f6539e93d/exporters/otlp/otlplog/otlploghttp/config.go#L186

File: CHANGELOG.md [hunk 2]
```diff
@@ -31,6 +31,10 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 - Global MeterProvider registration unwraps global instrument Observers, the undocumented Unwrap() methods are now private. (#5881)
 - Fix `go.opentelemetry.io/otel/exporters/prometheus` trying to add exemplars to Gauge metrics, which is unsupported. (#5912)
+- Fix `go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc` using env variable endpoint scheme when `WithEndpointURL` is set. (#5944)
```
Comment (pellared) [inline CHANGELOG.md:34]: How about something like

```suggestion
- Fix `WithEndpointURL` to always use a secure connection when an https URL is passed in `go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc`. (#5944)
```

**Review by pellared — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

**Pull request merged**